### PR TITLE
TST: Add Python 3.7 to CI testing

### PR DESCRIPTION
**Pull request by cclauss**
Add Python 3.7 to CI testing.  Note that on travis-ci it must run with sudo on xenial, see travis-ci/travis-ci#9069.

### CODE REVIEW

**Comment by charris** [comment]
Since we are already testing with 3.7-dev, I'm not convinced the travis issue still applies. Can you try simply removing the `-dev` bit?

**Comment by cclauss** [comment]
Did you read through the link in the commit message?  3.7 beta != 3.7 production release.

**Comment by charris** [comment]
Looks like you can also add it to `.appveyor.yml`, see #11600.

**Comment by mhvk** [comment]
@cclauss - thanks for noticing those issues with old ubuntu versions! This looks good to me.

A question - probably best addressed separately - is whether we should be moving some of the other tests over to xenial as well. (There is also a possible optimization that we set in astropy but (apparently) not in numpy: `ccache`.)

**Comment by cclauss** [comment]
travis-ci/travis-ci#9069 is still an OPEN issue because [Precise and Trusty are still the supported Linux versions](https://docs.travis-ci.com/user/languages/python) so I would avoid against moving to Xenial unless necessary.

**Comment by charris** [comment]
Please keep the 3.6 testing in `.appveyor.yml`. I arranged things as I did because we will be dropping 3.4 for 1.16 and 2.7 for 1.17.

**Comment by cclauss** [comment]
@charris Ready for your review...  Have I missed anything?

File: doc/Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Makefile for Sphinx documentation
 #
 
-PYVER = 3.6
+PYVER = 3.7
```
Comment (charris) [inline doc/Makefile]: This one should not be changed, 3.6 is widespread these days, will probably be supported for a couple of years, and so using it to build the documentation makes sense. OTOH, 3.7 is brand new and not yet included in many distros.

Comment (cclauss) [reply doc/Makefile]: So do we revert Circle CI to Py36?

Comment (charris) [reply doc/Makefile]: >So do we revert Circle CI to Py36?

Hmm, to tell the truth, I don't know ;) Did you try it with only one of the two changed? It cannot hurt to stick with 3.6 for a while.

Comment (cclauss) [reply doc/Makefile]: CircleCI change without the Makefile change is:
* https://circleci.com/gh/numpy/numpy/2660?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Comment (charris) [reply doc/Makefile]: Guess it matters, good to know.

**Comment by charris** [comment]
Thanks @cclauss .

**Pull request merged**